test/membrane_rtc_engine/engine_test.exs

```diff
@@ -308,6 +322,93 @@ defmodule Membrane.RTC.EngineTest do
     end
   end
 
+  describe "Engine linking test" do
+    test "engine properly links for both formats", %{rtc_engine: rtc_engine} do
+      test_format_dump_endpoint_id = "test-format-dump-endpoint"
+      raw_dump_endpoint_id = "raw-dump-endpoint"
+
+      test_format_path = Path.join(@output_dir, "test-format")
+      raw_path = Path.join(@output_dir, "raw")
+
+      File.mkdir_p!(test_format_path)
+      File.mkdir_p!(raw_path)
+
+      raw_endpoint = %DumpEndpoint{directory_path: raw_path, format: :raw, rtc_engine: rtc_engine}
+
+      test_format_endpoint = %DumpEndpoint{
+        directory_path: test_format_path,
+        format: :test_format,
+        rtc_engine: rtc_engine
+      }
+
+      :ok = Engine.add_endpoint(rtc_engine, raw_endpoint, endpoint_id: raw_dump_endpoint_id)
+
+      :ok =
+        Engine.add_endpoint(rtc_engine, test_format_endpoint,
+          endpoint_id: test_format_dump_endpoint_id
+        )
+
+      file_endpoint_id = "file-endpoint-id"
+
+      file_name = "video_baseline.h264"
+      file_path = Path.join(@fixtures_dir, file_name)
+
+      track_id = "multiple-formats-test-track-id"
+      stream_id = "multiple-formats-test-stream"
+
+      track =
+        Engine.Track.new(
+          :video,
+          stream_id,
+          file_endpoint_id,
+          :H264,
+          90_000,
+          [:test_format, :raw],
+          %FMTP{pt: 127},
+          id: track_id
+        )
+
+      parser = %Membrane.H264.FFmpeg.Parser{
+        framerate: {24, 1},
+        alignment: :nal
+      }
+
+      file_endpoint = %FileEndpoint{
+        rtc_engine: rtc_engine,
+        file_path: file_path,
+        track: track,
+        interceptor: fn link_builder ->
+          link_builder
+          |> Membrane.ParentSpec.to(:parser, parser)
+        end,
+        depayloading_filter: NaiveFilter
+      }
+
+      :ok = Engine.add_endpoint(rtc_engine, file_endpoint, endpoint_id: file_endpoint_id)
+
+      assert_receive %Message.MediaEvent{rtc_engine: ^rtc_engine, to: :broadcast, data: data}
+
+      assert %{
+               "type" => "tracksAdded",
+               "data" => %{
+                 "trackIdToMetadata" => %{track_id => nil},
+                 "peerId" => file_endpoint_id
+               }
+             } == Jason.decode!(data)
+
+      Engine.message_endpoint(rtc_engine, file_endpoint_id, :start)
+
+      Process.sleep(15_000)
```

Comment:
This is not a good idea. For very slow machines this test won't pass. In general, we should avoid using `Process.sleep` for waiting for something to end. 

Instead of using `Membrane.Serializer` you can create simple element that will send each received buffer to testing process. This way you can be sure that end elements receives buffers which means that linking was successful.